packages/ncids-js/src/components/nci-header/utils/search.ts

```diff
@@ -99,126 +76,24 @@ export class Search {
 	 */
 	private handleSearch(event: Event): void {
 		event.preventDefault();
-		if (!this.mobileActive) {
-			this.createOverlay();
-			this.mobileActive = true;
-		}
+		const searchContainer = <HTMLElement>(
+			document.querySelector('.nci-header-nav__secondary')
+		);
+		searchContainer?.classList?.add('search-focused');
```

Comment:
Missing code coverage for lines 82-97. 

I assume it's due to the optional chaining on lines 82 and 97. 

We _do_ need to change the code to fail better when search is missing (as currently the component breaks other js from running on the platform if search is missing) - but we'll need 85% branch coverage for this file to pass the PR. Line 82 does not require the optional chaining because of the type casting done on line 79

something like this is the quick fix
```typescript
		const searchContainer = <HTMLElement>(
			document.querySelector('.nci-header-nav__secondary')
		);
		searchContainer.classList.add('search-focused');
   ...
		searchContainer.classList.remove('search-focused');
```

the long fix is to fix the broken search code if the search is not found then add unit tests for a header without a search - but we can definitely add a bug tix for that

Reply:
oh, i noticed that the constructor sets `this.search` with the same selector. If we can, it is better to only query the document once instead of 3 times. is there a need to query it on every user interaction?

Reply:
I had no idea the optional chaining was messing with the tests, I guess that makes sense though. I've removed the optional chaining for now